NPE in AuthInstallerUserManagerPrefetchingImpl in AEMCloud with incomplete/invalid search index in place

With the most recent 2.7.2-SNAPSHOT I ran into the following exception

```
*INFO* [qtp9431062-822-ACTool-Config-Worker] biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthInstallerUserManagerPrefetchingImpl Prefetched 1 authorizables in 15ms
...
*ERROR* [qtp9431062-822] biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook Exception during execution of install hook: biz.netcentric.cq.tools.actool.authorizableinstaller.AuthorizableCreatorException: java.lang.NullPointerException
...
Caused by: biz.netcentric.cq.tools.actool.authorizableinstaller.AuthorizableCreatorException: java.lang.NullPointerException
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAuthorizables(AcInstallationServiceImpl.java:564)
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAcConfiguration(AcInstallationServiceImpl.java:343)
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installMergedConfigurations(AcInstallationServiceImpl.java:668)
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installConfigurationFiles(AcInstallationServiceImpl.java:304)
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installConfigurationFiles(AcInstallationServiceImpl.java:270)
	at biz.netcentric.cq.tools.actool.installhook.impl.AcToolInstallHookServiceImpl.installYamlFilesFromPackage(AcToolInstallHookServiceImpl.java:63)
	at biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook.install(AcToolInstallHook.java:99)
	... 73 common frames omitted
Caused by: java.lang.NullPointerException: null
	at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthInstallerUserManagerPrefetchingImpl.<init>(AuthInstallerUserManagerPrefetchingImpl.java:94)
	at biz.netcentric.cq.tools.actool.authorizableinstaller.impl.AuthorizableInstallerServiceImpl.installAuthorizables(AuthorizableInstallerServiceImpl.java:114)
	at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installAuthorizables(AcInstallationServiceImpl.java:551)
	... 79 common frames omitted

```

--- a/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthInstallerUserManagerPrefetchingImpl.java
+++ b/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthInstallerUserManagerPrefetchingImpl.java
@@ -97,6 +97,11 @@ class AuthInstallerUserManagerPrefetchingImpl implements AuthInstallerUserManage
                 Group memberOfGroup = declaredMemberOf.next();
                 String memberOfGroupId = memberOfGroup.getID();
                 isMemberOfByAuthorizableId.get(authId).add(memberOfGroupId);
+                if (!nonRegularUserMembersByAuthorizableId.containsKey(memberOfGroupId)) {
+                    throw new IllegalStateException("Group '" + memberOfGroupId
+                            + "' is not returned by query /jcr:root/home//element(*,rep:Authorizable)[(@jcr:primaryType!='rep:User' or @rep:authorizableId='anonymous')] "
+                            + "(as internally used by userManager.findAuthorizables()) - most likely there is a corrupt index on your platform.");
+                }
                 nonRegularUserMembersByAuthorizableId.get(memberOfGroupId).add(authId);
                 membershipCount++;
             }

